src/Mod/Part/App/WireJoiner.cpp

```diff
@@ -0,0 +1,2440 @@
+            for (auto &info : edges) {
+                if (info.iteration<0)
+                    continue;
+                for (int k=0; k<2; ++k) {
+                    int i;
+                    for (i=info.iStart[k]; i<info.iEnd[k]; ++i) {
+                        const auto &v = adjacentList[i];
+                        auto other = v.edgeInfo();
+                        if (other->iteration >= 0 && other != &info)
+                            break;
+                    }
+                    if (i == info.iEnd[k]) {
+                        // If merge or tight bound, then repeat until no edges
+                        // can be skipped.
+                        done = !doMergeEdge & !doTightBound;
+                        info.iteration = -3;
+                        showShape(&info, "skip");
+                        break;
+                    }
+                }
+            }
+        }
+    }
+
+    // This algorithm tries to find a set of closed wires that includes as many
+    // edges (added by calling add() ) as possible. One edge may be included
+    // in more than one closed wires if it connects to more than one edges.
+    void findClosedWires(bool tightBound=false)
+    {
+        std::unique_ptr<Base::SequencerLauncher> seq(
+                new Base::SequencerLauncher("Finding wires", edges.size()));
+
+        for (auto &info : edges) {
+            info.wireInfo.reset();
+            info.wireInfo2.reset();
+        }
+
+        for (auto it=edges.begin(); it!=edges.end(); ++it) {
+            VertexInfo beginVertex(it, true);
+            auto &beginInfo = *it;
+            seq->next(true);
+            ++iteration;
+            if (beginInfo.iteration < 0 || beginInfo.wireInfo)
+                continue;
+
+            VertexInfo currentVertex(it, true);
+            EdgeInfo *currentInfo = &beginInfo;
+            showShape(currentInfo, "begin");
+            stack.clear();
+            vertexStack.clear();
+            edgeSet.clear();
+
+            TopoDS_Wire wire = _findClosedWires(beginVertex, currentVertex);
+            if (wire.IsNull())
+                continue;
+
+            if (tightBound) {
+                assert(!beginInfo.wireInfo);
+                beginInfo.wireInfo.reset(new WireInfo);
+                beginInfo.wireInfo->vertices.emplace_back(it, true);
+                beginInfo.wireInfo->wire = wire;
+            }
+            for (auto &r : stack) {
+                const auto &v = vertexStack[r.iCurrent];
+                auto &info = *v.it;
+                if (tightBound)
+                    beginInfo.wireInfo->vertices.push_back(v);
+                if (!info.wireInfo) {
+                    info.wireInfo = beginInfo.wireInfo;
+                    // showShape(&info, "visited");
+                }
+            }
+            showShape(wire,"joined");
+            if (!tightBound)
+                builder.Add(compound, wire);
+        }
+    }
+
+    void checkStack()
+    {
+#if 0
```

Comment:
Should this code still be here?

Reply:
Maybe not.

Have you found cases like this in your PRs?
If so have you removed those parts of code?

Reply:
I have, and I have removed both the #if 0 and the OCCT version too old sections.  This looked like early debug code to me, but I wasn't sure.